docs/zkapps/tutorials/05-common-types-and-functions.mdx

```diff
@@ -258,11 +258,21 @@ const result = Circuit.if(
 
 Note that unlike normal programming, because SnarkyJS under the hood is creating a zk cirucit, this is in fact executing both branches and selecting the one that matches the condition.
 
-## Assertions
+## Assertions and Constraints
 
-We've been using assertions throughout these tutorials already - but to clarify what they're doing, assertions are creating constraints in our circuit, that must hold true for a valid proof to be generated.
+SnarkyJS functions are compiled to generate circuits.
 
-Besides the `a.assertEquals(b)` we've often used, there is also `.assertTrue()` available on the Bool class.
+When a transaction is proven in SnarkyJS, SnarkyJS is proving that the program logic is computed according to the written program, and all assertions are holding true.
+
+We've seen assertions already, with `a.assertEquals(b)` we've often used, there is also `.assertTrue()` available on the Bool class.
+
+Circuits in SnarkyJS currently have a fixed maximum size. Each operation performed in a function counts towards this maximum size. This maximum size is currently equivalent to the following:
+* about 5,200 hashes on two fields
+* about 2,600 hashes on four fields
```

Comment:
Should this say `Poseidon hashes` for clarity? Seems like a minor issue, will merge for now and you can PR it if you agree